src/OSX/Avalonia.MonoMac/WindowImpl.cs

```diff
@@ -0,0 +1,96 @@
+﻿using System;
+using Avalonia.Controls;
+using Avalonia.Platform;
+using MonoMac.AppKit;
+
+namespace Avalonia.MonoMac
+{
+    class WindowImpl : WindowBaseImpl, IWindowImpl
+    {
+        bool _decorated = true;
+
+        public WindowImpl()
+        {
+            UpdateStyle();
+        }
+
+        public WindowState WindowState
+        {
+            get
+            {
+                if (Window.IsMiniaturized)
+                    return WindowState.Minimized;
+                return WindowState.Normal;
+
+            }
+            set
+            {
+                if (value == WindowState.Maximized)
+                {
+                    if (Window.IsMiniaturized)
+                        Window.Deminiaturize(Window);
+                    if (!Window.IsZoomed)
+                        Window.PerformZoom(Window);
+                }
+                else if (value.HasFlag(WindowState.Minimized))
+                    Window.Miniaturize(Window);
+                else
+                {
+                    if (Window.IsMiniaturized)
+                        Window.Deminiaturize(Window);
+                    if (Window.IsZoomed)
+                        Window.IsZoomed = false;
+                }
+            }
+        }
+
+        public void SetIcon(IWindowIconImpl icon)
+        {
+            //No-OP, see http://stackoverflow.com/a/7038671/2231814
+        }
+
+        public void ShowTaskbarIcon(bool value)
+        {
+            //No-OP, there is no such this as taskbar in OSX
```

Comment:
I was thinking this would map over as turning on and off having an icon visible in the Dock since that would make the most sense.

Reply:
Dock icon isn't tied to particular window, it's application-global. OSX will show dock icon even if there are no windows.